src/main/org/codehaus/groovy/util/ManagedReference.java

```diff
@@ -46,7 +46,6 @@ public final T get() {
     
     public final void clear() {
         ref.clear();
-        manager.removeStallEntries();
```

Comment:
`ClassInfo#finalizeReference` calls `setStrongMetaClass(null)` which in turn calls `replaceWeakMetaClassRef(null)` which calls `weakRef#clear` which is on ManagedReference.  This causes a recursive call back into the queue can can lead to stackoverflow if there are lots of ClassInfo ManagedReferences in the Reference Queue.

A related PR that attempts to prevent this reentrant processing is PR #298.